### PR TITLE
Update GDB settings to enable ITM, add example

### PR DESCRIPTION
**Pull request by maxekman**
This adds the correct ITM configurations and an example taken from one of the other STM32 crates. It also makes the GDB and OpenOCD files more similar to some of the other crates.

I needed ITM for developing the I2S proposal (https://github.com/rust-embedded/embedded-hal/pull/204) to enable the DAC capability for this BSC eventually.

### CODE REVIEW

File: .cargo/config [hunk 1]
```diff
@@ -1,7 +1,28 @@
-[target.thumbv7em-none-eabihf]
-runner = 'arm-none-eabi-gdb'
+[target.thumbv7m-none-eabi]
+# uncomment this to make `cargo run` execute programs on QEMU
+# runner = "qemu-system-arm -cpu cortex-m3 -machine lm3s6965evb -nographic -semihosting-config enable=on,target=native -kernel"
+
```
Comment (therealprof) [inline .cargo/config:4]: Can we get rid of these?

File: .cargo/config [hunk 2]
```diff
@@ -1,7 +1,28 @@
...
+  # if you run into problems with LLD switch to the GNU linker by commenting out
+  # this line
+  # "-C", "linker=arm-none-eabi-ld",
+
+  # if you need to link to pre-compiled C libraries provided by a C toolchain
+  # use GCC as the linker by commenting out both lines above and then
+  # uncommenting the three lines below
+  # "-C", "linker=arm-none-eabi-gcc",
+  # "-C", "link-arg=-Wl,-Tlink.x",
+  # "-C", "link-arg=-nostartfiles",
```
Comment (therealprof) [inline .cargo/config:25]: And these?

**Comment by therealprof** [comment]
The rest looks good to me.

File: .cargo/config (hunk 1 above)
Comment (maxekman) [reply .cargo/config:4]: Sure, sounds ok to me. Haven’t tested it either, the QEMU settings all came from one of the other BSCs where I got this from. Will clean it up later today.

File: .cargo/config [hunk 3]
```diff
@@ -1,5 +1,8 @@
-[target.thumbv7em-none-eabihf]
-runner = 'arm-none-eabi-gdb'
+[target.thumbv7m-none-eabi]
+
```
Comment (therealprof) [inline .cargo/config:2]: This one too, please. ;)

Comment (maxekman) [reply .cargo/config]: My bad, thought it made sense to leave it. Removed now.

**Pull request merged**